apps/web/pages/opengraph/post/[postId].tsx

```diff
@@ -0,0 +1,111 @@
+import { useRouter } from 'next/router';
+import { useMemo } from 'react';
+import { graphql, useLazyLoadQuery } from 'react-relay';
+
+import { PostOpenGraphPreview } from '~/components/opengraph/PostOpenGraphPreview';
+import { HEIGHT_OPENGRAPH_IMAGE, WIDTH_OPENGRAPH_IMAGE } from '~/constants/opengraph';
+import { PostIdOpengraphQuery } from '~/generated/PostIdOpengraphQuery.graphql';
+import getVideoOrImageUrlForNftPreview from '~/shared/relay/getVideoOrImageUrlForNftPreview';
+
+export default function OpenGraphPostPage() {
+  const { query } = useRouter();
+  const queryResponse = useLazyLoadQuery<PostIdOpengraphQuery>(
+    graphql`
+      query PostIdOpengraphQuery($postId: DBID!) {
+        post: postById(id: $postId) {
+          ... on ErrPostNotFound {
+            __typename
+          }
+          ... on Post {
+            __typename
+            author @required(action: THROW) {
+              username
+              profileImage {
+                ... on TokenProfileImage {
+                  token {
+                    ...getVideoOrImageUrlForNftPreviewFragment
+                  }
+                }
+                ... on EnsProfileImage {
+                  __typename
+                  profileImage {
+                    __typename
+                    previewURLs {
+                      medium
+                    }
+                  }
+                }
+              }
+            }
+            caption
+            tokens {
+              ...getVideoOrImageUrlForNftPreviewFragment
+            }
+          }
+        }
+      }
+    `,
+    { postId: query.postId as string }
+  );
+  const { post } = queryResponse;
+
+  // stripped down version of the pfp retrieving logic in ProfilePicture.tsx
+  const profileImageUrl = useMemo(() => {
+    if (!post || post?.__typename !== 'Post') {
+      return null;
+    }
+
+    const { token, profileImage } = post.author.profileImage ?? {};
+
+    if (profileImage && profileImage.previewURLs?.medium) {
+      return profileImage.previewURLs.medium;
+    }
+
+    if (token) {
+      const result = getVideoOrImageUrlForNftPreview({
+        tokenRef: token,
+        handleReportError: reportError,
+      });
+      return result?.urls.small ?? null;
+    }
+
+    return null;
+  }, [post]);
```

Comment:
awesome- depending on how often we do this could probably be a util

Reply:
yess tho hopefully not too often! ProfilePicture should be able to handle most cases  so thinking OpenGraph is a limited case where we don't want it to need to do as much to display the image 🤞